crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+
+type GasCoin {}
+type Input   { ix: Int }
+type Result  { cmd: Int, ix: Int }
+
+union ProgrammableTransaction =
+    MoveCallTransaction
+  | TransferObjectsTransaction
+  | SplitCoinTransaction
+  | MergeCoinsTransaction
+  | PublishTransaction
+  | UpgradeTransaction
+  | MakeMoveVecTransaction
+
+type MoveCallTransaction {
+  moduleId: MoveModuleId
+  name: String
+  function: NormalizedFunction
+
+  typeArguments: [MoveType]
+  arguments: [TransactionArgument]
+}
+
+type TransferObjectsTransaction {
+  objects: [TransactionArgument]
+  address: TransactionArgument
+}
+
+type SplitCoinTransaction {
+  coin: TransactionArgument
+  amounts: [TransactionArgument]
+}
+
+type MergeCoinsTransaction {
+  coin: TransactionArgument
+  coins: [TransactionArgument]
+}
+
+type PublishTransaction {
+  dependencies: [MovePackage]
+}
+
+type UpgradeTransaction {
+  dependencies: [MovePackage]
+  currentPackage: MovePackage
+  upgradeTicket: TransactionArgument
+}
+
+type MakeMoveVecTransaction {
+  type: MoveType
+  elements: [TransactionArgument]
+}
+
+# Omitting for brevity -- there's a lot of structure here, but it's
+# not very "connected" (i.e. no edges to other types).
+type TransactionSignature {}
+
+type TransactionBlockEffects {
+  digest: Digest!
+  status: ExecutionStatus!
+  errors: String
+  transactionBlock: TransactionBlock
+  dependencies: [TransactionBlock]
+
+  lamportVersion: U64
+  gasEffects: GasEffects
+  objectReads: [Object]
+  objectChanges: [ObjectChange]
+  balanceChanges: [BalanceChange]
+
+  epoch: Epoch
+  checkpoint: Checkpoint
+
+  events: TransactionEventConnection
+
+  bcs: Bytes
+}
+
+enum ExecutionStatus {
+  SUCCESS
+  FAILURE
+}
+
+type GasInput {
+  gasSponsor: Address
+  gasPayment: [Object]
+
+  gasPrice: U64
+  gasBudget: U64
+}
+
+type GasEffects {
+  gasObject: Coin
+  gasSummary: GasCostSummary
+}
+
+type GasCostSummary {
+  computationCost: U64
+  storageCost: U64
+  storageRebate: U64
+  nonRefundableStorageFee: U64
+}
+
+type ObjectChange {
+  inputState: Object
+  outputState: Object
+
+  idCreated: Boolean
+  idDeleted: Boolean
+}
+
+type BalanceChange {
+  owner: Owner
+  coinType: MoveType
+  amount: I128
+}
+
+type Event {
+  moduleId: MoveModuleId
+  type: MoveType
+  sender: Address
+  timestampMs: U64
+
+  json: JSON
+  bcs: Bytes
+}
+
+type Balance {
+  coinType: MoveType
+  coinObjectCount: U64
+  totalBalance: U128
+}
+
+type Coin {
+  balance: U64
+  asMoveObject: MoveObject
+}
+
+type StakedSui {
+  status: StakeStatus
+  requestEpoch: Epoch
+  activeEpoch: Epoch
+  principal: U64
+
+  # Only available if status is `ACTIVE`.
+  estimatedReward: U64
+
+  asMoveObject: MoveObject
+}
+
+enum StakeStatus {
+  PENDING
+  ACTIVE
+  UNSTAKED
+}
+
+type CoinMetadata {
+  decimals: Int
+  name: String
+  symbol: String
+  description: String
+  iconURL: String
+  supply: U64
+
+  asMoveObject: MoveObject
+}
+
+input DynamicFieldName {
+  type: TypeInst!
+  bcs: Bytes!
```

Comment:
This implies that the new dynamic field querying API requires the client to serialise the field name into BCS, is that an okay assumption to make?